misk-feature/src/main/kotlin/misk/feature/ValidatableConfig.kt

```diff
@@ -0,0 +1,10 @@
+package misk.feature
+
+import misk.config.Config
+
+/**
+ * Interface used by [ConsumerConfigResolver] to support validation of retrieved configs.
+ */
+interface ValidatableConfig<T> : Config {
+  fun validate()
```

Comment:
what are semantics of this method? does it throw? what does it throw?

Reply:
Is it too loose to just suggest IllegalState/IllegalArgument exceptions?

Reply:
i think throwing IllegalArgumentException is fine.